### DISCUSSION TITLE
Missing group_by in LATERAL correlation example

### DISCUSSION THREAD

**Discussion by jonbiemond** [sqlalchemy/sqlalchemy#12090] [Ideas]
### Describe the bug

The example in the documentation for LATERAL correlation is missing a `group_by` clause.

Page: https://docs.sqlalchemy.org/en/20/tutorial/data_select.html#lateral-correlation
GitHub: https://github.com/sqlalchemy/sqlalchemy/blob/fa76fbda4aa2887d001e1e15f9bfe9dfb24fadc8/doc/build/tutorial/data_select.rst?plain=1#L1082-L1090

### Proposed solution
```python
subq = (
    select(
        func.count(address_table.c.id).label("address_count"),
        address_table.c.email_address,
        address_table.c.user_id,
    )
    .group_by(address_table.c.email_address, address_table.c.user_id)
    .where(user_table.c.id == address_table.c.user_id)
    .lateral()
)
```

### Optional link from https://docs.sqlalchemy.org which documents the behavior that is expected

https://docs.sqlalchemy.org/en/20/tutorial/data_select.html#lateral-correlation

### SQLAlchemy Version in Use

2.0.36

### DBAPI (i.e. the database driver)

psycopg

### Database Vendor and Major Version

PostgreSQL 17.0

### Python Version

3.10

### Operating system

Windows

### To Reproduce

```python
"""
CREATE TABLE user_account (
    id SERIAL PRIMARY KEY,
    name VARCHAR
);

CREATE TABLE address (
    id SERIAL PRIMARY KEY,
    user_id INTEGER REFERENCES user_account(id),
    email_address VARCHAR
);

INSERT INTO user_account (id, name)
VALUES (1, 'sandy'), (2, 'patrick');

INSERT INTO address (user_id, email_address)
VALUES
    (1, 'sandy@sqlalchemy.org'),
    (1, 'sandy@sqlalchemy.org'),
    (2, 'patrick@mail.com');

SELECT user_account.name, anon_1.address_count, anon_1.email_address
FROM user_account
JOIN LATERAL (SELECT count(address.id) AS address_count,
address.email_address AS email_address, address.user_id AS user_id
FROM address
WHERE user_account.id = address.user_id) AS anon_1
ON user_account.id = anon_1.user_id
ORDER BY user_account.id, anon_1.email_address;
"""
```

### Error

```
ERROR: column "address.email_address" must appear in the GROUP BY clause or be used in an aggregate function
```

**Comment by jonbiemond** [comment]
Happy to open a PR, if confirmed as an issue.

**Comment by zzzeek** [comment]
that example in the tutorial is only trying to illustrate how to apply the LATERAL keyword and not actually running the query.   Im not sure the query makes any sense, if you grouped by the other columns you'd always get "1" for address_count and it doesnt really do anything that would require LATERAL anyway.

if we want to go through the effort to change this query I'd seek to look at the better examples of LATERAL in the PG docs and try to write a new query based on that.    (my own SQL experience has never taken me to need something like LATERAL)

**Comment by zzzeek** [comment]
we dont need issues open for small documentation edits
